### PR TITLE
[NETBEANS-862] Added auto-complete support for var keyword in JDK-11 Lambda Parameters

### PR DESCRIPTION
**Pull request by arusinha**
Added Auto-complete support  for var keyword in Lambda parameter (JDK-11 or above). 

e.g.,(Suppose cursor is at '^' position. 
   IntBinaryOperator calc2 = (var x, ^ y ) -> x+y;

on ctl+space, below expected outcome should happen.
 IntBinaryOperator calc2 = (var x, var y ) -> x+y;

Test cases needs to be re-verified once nb-javac for jdk-11 is available.

Jira Link: https://issues.apache.org/jira/browse/NETBEANS-862

### CODE REVIEW

File: java.completion/src/org/netbeans/modules/java/completion/JavaCompletionTask.java [hunk 1]
```diff
@@ -1903,6 +1909,11 @@ private void insideLambdaExpression(Env env) throws IOException {
                         addPrimitiveTypeKeywords(env);
                         addKeyword(env, FINAL_KEYWORD, SPACE, false);
                     }
+                    //Assuming first parameter to be var type if position for the parameter type is not found.
+                    else if (!let.getParameters().isEmpty() && let.getParameters().get(0).getType() != null )
```
[the remark this answers is not in the document]
Comment (arusinha) [reply java.completion/src/org/netbeans/modules/java/completion/JavaCompletionTask.java]: formatting issue fixed in last commit. Also first condition check was redundant , is removed now

File: java.completion/src/org/netbeans/modules/java/completion/JavaCompletionTask.java [hunk 2]
```diff
@@ -5868,4 +5880,32 @@ private boolean withinBounds(Env env, TypeMirror type, List<? extends TypeMirror
...
+        final CompilationController controller = env.getController();
+        final Types types = controller.getTypes();
+        final Elements elements = controller.getElements();
+        Set<? extends TypeMirror> smartTypes = getSmartTypes(env);
+        if (smartTypes != null) {
+            for (TypeMirror st : smartTypes) {
+                if (st.getKind().isPrimitive()) {
+                    st = types.boxedClass((PrimitiveType) st).asType();
+                }
+                if (st.getKind() == TypeKind.DECLARED) {
```
[the remark this answers is not in the document]
Comment (arusinha) [reply java.completion/src/org/netbeans/modules/java/completion/JavaCompletionTask.java]: First condition check was redundant , is removed now. So no need for else

**Comment by rtaneja1** [comment]
Please check Jan's PR https://github.com/apache/incubator-netbeans/pull/616 if any changes required in tests.

**Comment by arusinha** [comment]
PR #616 is not affecting  my PR changes , I had verified

**Comment by rtaneja1** [comment]
This needs to be tested with updated nb-javac(with JDK 11 uptake).

**Comment by arusinha** [comment]
Will create fresh review request for thew issue

**Pull request closed without merging**